src/components/Courses/Courses.jsx

```diff
@@ -43,23 +43,15 @@ const Courses = () => {
 	const [curInput, setCurInput] = useState('');
 	const courses = useSelector((state) => state.courses.courses);
 	const authors = useSelector((state) => state.authors.authors);
+	const admin = useSelector((state) => state.authentication.role);
+	const token = useSelector((state) => state.authentication.token);
 	const dispatch = useDispatch();
 
 	useEffect(() => {
-		async function fetchData() {
-			const data = await fetchDataGo('courses/all');
-			dispatch(actionCreators.setCourses(data));
-		}
-		fetchData();
-	}, [dispatch]);
-
-	useEffect(() => {
-		async function fetchData() {
-			const data = await fetchDataGo('authors/all');
-			dispatch(actionCreatorsAuthors.setAuthors(data));
-		}
-		fetchData();
-	}, [dispatch]);
+		dispatch(getCoursesThunk());
+		dispatch(getAuthorsThunk());
+		dispatch(setRoleThunk(token));
```

Comment:
I'd recommend to set user role at root level of your application: you may need role not only in courses list